### ISSUE TITLE
[New azd-template] Flask surveys container app

### ISSUE THREAD

**Issue by pamelafox** [Azure/awesome-azd#88]
## New azd-template Submission
Fill in the sections below to populate a new gallery card for your template. **All fields are mandatory unless explicitly identified as optional**

## 1. Description
_1-2 sentence description of the architecture or solution that is defined by the template. If this is a requested template, please tag the GitHub issue containing the request._

A survey app written with the Python Flask framework which uses Flask-SQLAlchemy with a PostgreSQL database. Can be developed locally with Docker Compose and then deployed to Azure Container Apps (using KeyVault SDK for Flask secrets).

## 2. Author Information
**Name:** Pamela Fox

**Link:** https://github.com/pamelafox

## 3. Template Repo
_The GitHub repo containing the template being contributed-- must be a valid, complete template_

https://github.com/pamelafox/flask-surveys-container-app

## 4. Link to PR where you made your repo `azd` compatible
_To allow us to provide feedback on your template, please include a link to the PR where you added `azd` compatibility_

Sorry, there's not one particular PR, but you can check azure.yaml, the infra/ folder, and .github/workflows.

## 5. Tags
**Standard:** _One or more tags representing the template. Look at the currently available tags on the [Gallery page](https://azure.github.io/awesome-azd/). Provide **at least 1 tag** for programming language used and **at least 1 tag for Azure services integrated**. Also tag the **IaC provider** (Bicep or Terraform)_

Python, Flask, Azure Container Apps, Azure Key Vault, Web Apps, Azure DB for PostgreSQL

**Additional:** _Don't see a relevant tag for your template? Suggest tags to be added and used for your template. These could be new languages, services, or application domains_.

## 6. Architecture Diagram or Application Screenshot
_Used as display image for gallery card-- should include all services and their connections. [Example](https://github.com/Azure-Samples/todo-csharp-sql/blob/main/assets/resources.png)_

![flaskdb_container drawio (1)](https://user-images.githubusercontent.com/297042/216431698-a128ee54-517b-40f7-801c-fbcb0216e550.png)

## 7. Related Issues (optional)
_Is this a template that was requested in a separate issue? If so, please tag the related issue here._
---

## Next Steps
Once submitted, the issue will be reviewed - we plan to do reviews on a rolling basis at regular intervals. The process will include verifying all information required for the template gallery is provided and the template works (i.e., successfully deploys to Azure with `azd up`). 
 * If we have questions or enhancements, we will add comments in issue thread (**issue stays open**)
 * If the contribution is approved, we'll update the gallery directly (**issue will then be closed**)

**Comment by rajeshkamal5050** [comment]
@v-xuto can you do a dry-run on this template and update here for any issues/feedback?

**Comment by v-xuto** [comment]
@pamelafox We tested this template, encountered an issue, and filed this issue (https://github.com/pamelafox/flask-surveys-container-app/issues/5). 
Please help to check and solve it.

**Comment by pamelafox** [comment]
Replied there, thanks!

**Comment by v-xuto** [comment]
@jongio We have run this template without any issues. Please review it.

**Comment by pamelafox** [comment]
(Writing similar comment here as I wrote on my other ACA submission)

After a discussion with some App Service and ACA PMs last week, I've realized that websites aren't the most appropriate use of ACA. A website should be hosted as a BYOC Web App, which azd does not yet support (https://github.com/Azure/azure-dev/issues/1608). 
This article goes more into optimal uses of ACA:
https://learn.microsoft.com/en-us/azure/container-apps/compare-options

So my proposal is to table this template until azd supports BYOC, and I will submit another template instead that's more of a service than a website. The infrastructure will be largely the same.

**Comment by jongio** [comment]
As there are many tradeoffs when it comes to consider where to host an app, and it being possible to host a site on ACA, then I don't think it does any harm to create a sample showing a website hosted on ACA.  I 1000% guarantee a customer will try it.  In addition, a website may be a small portion of a complete ACA solution, that includes a site, api, db, etc.

**Comment by pamelafox** [comment]
True. For me, the primary downside is people not realizing that App Service BYOC is also an option (I keep forgetting about it myself). You're welcome to put this in the templates gallery if you think it's useful.
I still would like to build an API-oriented template, using PostGIS on top of PostgreSQL, but haven't built that one yet.

**Comment by rajeshkamal5050** [comment]
@jongio @pamelafox do we want to add this template to gallery or hold off?

**Comment by pamelafox** [comment]
I'm good with adding it now, I realize that ACA is a compelling option due to the way it's billed. (I did a price comparison of my own ACA/AppService apps which helped me realize that- http://blog.pamelafox.org/2023/03/hosting-python-web-apps-on-azure-price.html)

**Comment by rajeshkamal5050** [comment]
@jongio please review when you get a chance.

@hemarina please add it to gallery post Jon's approval.

**Comment by jongio** [comment]
:shipit:

**Comment by rajeshkamal5050** [comment]
@hemarina please add it to the gallery
